### PR TITLE
feat: upgrade to carbon 11

### PR DESCRIPTION
**Pull request by Akshat55**
Upgrade for carbon v11

#### Changelog

**New**
* Deprecates node-sass for sass
* Deprecates carbon-components for @carbon/styles
* Adds carbon 11 support

**Changed**
* Updates all carbon packages
* Updates to new `cds--` prefix from `bx--`
  * Will need to go over each component to ensure it is working (Tooltip has been reworked)
  * Remove anything that is NOW deprecated

**Removed**
* carbon-components dependency

### CODE REVIEW

**Review by zvonimirfras — CHANGES_REQUESTED** [review]

File: package.json [hunk 1]
```diff
@@ -64,7 +69,7 @@
     "@angular/forms": "^7.0.0 || ^8.0.0 || ^9.0.0 || ^10.0.0 || ^11.0.0 || ^12.0.0 || ^13.0",
     "rxjs": "^6.0.0",
     "zone.js": "^0.8.26 || ^0.9.0 || ^0.10.0",
-    "carbon-components": "^10.0.0"
+    "@carbon/styles": "^0.16.0"
```
[the remark this answers is not in the document]
Comment (Akshat55) [reply package.json:72]: And the start of a new one 1️⃣1️⃣

File: src/icon/icon.module.ts [hunk 2]
```diff
@@ -106,7 +105,6 @@ export class IconModule {
 			Copy20,
 			Data216,
 			Data220,
-			Delete16,
```
Comment (Akshat55) [inline src/icon/icon.module.ts:109]: V11 removes Delete icon & the Trash Can icon is its replacement, so should I import that instead?

Comment (zvonimirfras) [reply src/icon/icon.module.ts:109]: And replace it wherever is used (table story)

File: src/icon/icon.service.ts [hunk 3]
```diff
@@ -16,14 +16,14 @@ import {
...
 	ErrorFilled16,
 	InformationFilled16,
 	Menu16,
 	OverflowMenuVertical16,
 	Save16,
 	Settings16,
+	TrashCan16,
 	Warning16,
 	WarningFilled16
 } from "@carbon/icons";
```
Comment (Akshat55) [inline src/icon/icon.service.ts:29]: All imports in imports declaration are unused 🤔 Do we really need this?

File: src/table/table.stories.ts [hunk 4]
```diff
@@ -182,15 +184,15 @@ storiesOf("Components|Table", module).addDecorator(
 					<ibm-table-toolbar-actions>
 						<button ibmButton="primary" [tabindex]="toolbar.selected ? 0 : -1">
 							Delete
-							<svg ibmIcon="delete" size="16" class="bx--btn__icon"></svg>
+							<div ibmIcon="trash-can" size="16" class="cds--btn__icon"></div>
```
Comment (zvonimirfras) [inline src/table/table.stories.ts:187]:
```suggestion
							<svg ibmIcon="trash-can" size="16" class="cds--btn__icon"></svg>
```

**Pull request merged**